source/common/network/tcp_listener_impl.cc

```diff
@@ -82,12 +82,15 @@ void TcpListenerImpl::onSocketEvent(short flags) {
     // Pass the 'v6only' parameter as true if the local_address is an IPv6 address. This has no
     // effect if the socket is a v4 socket, but for v6 sockets this will create an IPv4 remote
     // address if an IPv4 local_address was created from an IPv6 mapped IPv4 address.
-    const Address::InstanceConstSharedPtr& remote_address =
-        (remote_addr.ss_family == AF_UNIX)
-            ? io_handle->peerAddress()
-            : Address::addressFromSockAddr(remote_addr, remote_addr_len,
-                                           local_address->ip()->version() ==
-                                               Address::IpVersion::v6);
+    Address::InstanceConstSharedPtr remote_address;
+    if (remote_addr.ss_family == AF_UNIX) {
+      remote_address = io_handle->peerAddress();
+    } else {
+      StatusOr<Address::InstanceConstSharedPtr> error_or_address = Address::addressFromSockAddr(
+          remote_addr, remote_addr_len, local_address->ip()->version() == Address::IpVersion::v6);
+      RELEASE_ASSERT(error_or_address.ok(), error_or_address.status().ToString());
```

Comment:
I worry when I see a RELEASE_ASSERT where there was not one previously. Do we instead need to propagate an error up and replace a 'try/catch' with an error-check? How do we know there isn't a try/catch in the call-stack here?

Reply:
Yeah I think if we have `ASSERT(!IsMainThread())` that will help give us context we will not be trying to catch an exception for this error. The main concern then is that you might get a false-positive in thread-free unit tests.